### ISSUE TITLE
Expose something finer than just `.save()` through the datastore abstraction

### ISSUE THREAD

**Issue by chilts** [GoogleCloudPlatform/gcloud-node#384]
One thing I'd like is more features than just `.save()`. At the moment, I'm inserting a kind=User with name=chilts. However if I were to have another user ask for the name 'chilts', then `.save()` will say "Oh, the key is complete, so I'll do an update". I would like to make it try an insert and fail because that name already exists (ie. the primary key) ... is that currently possible via your api (from what I can tell it isn't)?

I love how all of the thorny details are abstracted away, but possibly exposing more of the API through here would be awesome. i.e. `.insert()`, `.update()` as well as `.save()`. Sometimes the key logic in `.save()` isn't needed.

Thanks again for a great library.

**Comment by dhermes** [comment]
@chilts The [relevant][1] verbs are:

- `update` - Must exist
- `insert` - Must not exist
- `upsert` - Can be brand new or exist already
- `insert_auto_id` - Incomplete path

I am also interested in

1. The demand for this from end users
1. The solution to this demand

[1]: https://github.com/GoogleCloudPlatform/google-cloud-datastore/blob/21e029b2657ebb15fef5429c90c4315326855fd1/proto/datastore_v1.proto#L388-L400

**Comment by dhermes** [comment]
ISTM after a quick scan that the [relevant bits of code][1] only use `upsert` and `insert_auto_id` (as is done in `gcloud-python`).

It just build an accumulator which starts out as

```javascript
    {
      upsert: [],
      insert_auto_id: []
    }
```

and is translated to the relevant protobuf via `DatastoreRequest.prototype.makeReq_`.

You could build your own accumulator with `update` and `insert` keys and then execute

```
DatastoreRequest().makeReq_('commit', req, onCommitCallback);
```

[1]: https://github.com/GoogleCloudPlatform/gcloud-node/blob/d73ce95ee552bf4e0d0edd711e4200cfa70cb5a4/lib/datastore/request.js#L255-L260

**Comment by ryanseys** [comment]
Agreed, we should support `insert` functionality on its own as it's pretty standard for the "attempting to add new users" use-case that @chilts described above.

**Comment by ryanseys** [comment]
@chilts How does #387 fill this need? Basically you'll explicitly set a `method` for your entity like so:

``` js
var gcloud = require('gcloud');
var datastore = gcloud.datastore;
var dataset = datastore.dataset();

dataset.save([
  {
    key: dataset.key({ path: ['User', 'chilts'] }),
    method: 'insert', // force an insert
    data: {
      updateTime: (new Date()).toString()
    }
  }
], function(err, result) {
  console.log(err, result);
});
```

**Comment by chilts** [comment]
Yeah, that would probably work.

For now (for my current project) I've moved to a different datastore so I'll just keep my eye on how you guys progress over here. Cheers.

**Comment by ryanseys** [comment]
Cool please let us know if you have any more ideas how we can make the library better, your feedback is very important :)

**Comment by stephenplusplus** [comment]
This is now merged in and will make it for 0.13.0.
